src/components/commentsForm.tsx

```diff
@@ -0,0 +1,87 @@
+import { FC, useState, useEffect } from 'react'
+import { Button, Box } from '@mui/material';
+
+import { CommentsFormInput } from './CommentsFormInput';
+
+type CommentsFormProps = {
+  addComments: (comments: any[]) => void,
+  initialList?: any[]
+};
+
+export const CommentsForm: FC<CommentsFormProps> = ({
+  addComments,
+  initialList,
+}) => {
+  const [isList, setIsList] = useState(initialList ? true : false)
+  const setId = () => (Math.round(Math.random() * 100000)).toString()
+
+  const [list, setList] = useState<any>(
+    initialList
+    ? initialList.map((i) => ({
+        id: setId(),
+        value: i
+      }))
+    : []
+  )
+
+  const addItem = () => {
+    setList([...list, {
+      id: setId(), 
+      value: '',
+    }])
+  }
+
+  const removeInput = (id: string) => {
+    setList(list.filter((i:{id: string, value: string}) => i.id !== id))
+  }
+
+  const saveValue = (id: string, value: string) => {
+    setList(list.map((i:{id: string, value: string}) => {
+      if(i.id === id) {
+        i.value = value
+      }
+
+      return i
+    }))
+  }
+
+  useEffect(() => addComments(list.map((i:{id: string, value: string}) => i.value)), [list])
+
+  if(!isList || list.length == 0) {
```

Comment:
also looks redundant, we can rename 'One more comment' to 'Add comment' in bottom section and use only bottom one